setup.py

```diff
@@ -245,11 +245,14 @@ def run(self):
     'watchdog>=0.10.3'
 ]
 
-# Replace spyder-kernels constraint to enable
-# building Windows installers on PRs
-if 'dev' in __version__ and WINDOWS_INSTALLER_NAME:
+# Loosen constraints to ensure dev versions still work
+if 'dev' in __version__:
+    install_requires.remove('python-lsp-server[all]>=1.4.1,<1.5.0')
+    install_requires.remove('qtconsole>=5.3.0,<5.4.0')
     install_requires.remove('spyder-kernels>=2.3.0,<2.4.0')
-    install_requires.append('spyder-kernels>=2.3.0,<=2.4.0.dev0')
+    install_requires.append('python-lsp-server[all]>=1.4.1,<1.6.0')
+    install_requires.append('qtconsole>=5.3.0,<5.5.0')
+    install_requires.append('spyder-kernels>=2.3.0,<2.5.0')
```

Comment:
Following this an update at https://github.com/spyder-ide/spyder/blob/5.x/RELEASE.md#update-core-dependencies is needed (the bullet point for `setup.py`)

Reply:
@dalthviz, I'm afraid I don't understand what you're suggesting. Could you elaborate?
@CAM-Gerlach, do we need something for `qdarkstyle` as well?

Reply:
Oh sorry about that 😅 @mrclary The idea is to update the text related to `setup.py` when doing a release (to not forget to also update this section). Currently the text from `RELEASE.md` says something like:

```
* Update the version of any packages required before the release in the following files:
   *  setup.py (look up for the Windows installer patch too)
```

But now probably should say something like:

```
* Update the version of any packages required before the release in the following files:
   *  setup.py (look up for the `install_requires` variable and also for the `Loosen constraints to ensure dev versions still work` patch )

```

Reply:
I was thinking about that; if we just do something like 

```python
if 'dev' in __version__:
    reqs_to_loosen = {'python-lsp-server[all]', 'qtconsole', 'spyder-kernels'}
    install_requires = [req for req in install_requires if req.split(">")[0] not in reqs_to_loosen]
    install_requires.append('python-lsp-server[all]>=1.4.1,<1.6.0')
    install_requires.append('qtconsole>=5.3.0,<5.5.0')
    install_requires.append('spyder-kernels>=2.3.0,<2.5.0')
```

This is both shorter/simpler and doesn't require manual updates, except for the dev deps after the release.

Reply:
But the update to `RELEASE.md` should still be the same, correct?

Reply:
Yeah, I guess so, though it in practice it requires less practical work updating, and doing so is less critical